torch/__init__.py

```diff
@@ -911,6 +911,63 @@ def is_warn_always_enabled():
     """
     return _C._get_warnAlways()
 
+
+def _check_with(error_type, cond, *args):
+    if not isinstance(cond, builtins.bool):
+        raise TypeError(f'cond must be a bool, but got {type(cond)}')
+
+    if cond:
+        return
+
+    # error_type must be a subclass of Exception and not subclass of Warning
+    assert issubclass(error_type, Exception) and not issubclass(error_type, Warning)
+
+    if len(args) == 0:
+        message = (
+            'Expected cond to be True, but got False. (Could this error '
+            'message be improved? If so, please report an enhancement request '
+            'to PyTorch.)')
+
+    else:
+        args_evaluated = []
+
+        for arg in args:
+            if callable(arg) and arg.__name__ == '<lambda>':
```

Comment:
Huh, testing the `__name__` being lambda is a strange thing to do

Reply:
I think I was following this: https://stackoverflow.com/a/24578359/14968956

I wasn't sure if we'd want to evaluate non-lambda callables too. If we do, I can get rid of the name check

Reply:
running non-lambda callable sounds good to me.
Also do we want to allow multiple arguments right now? Or do we just want to allow a single collable for now? (we can always allow more args in the future if we have a use case for it)